chat/chat.py

```diff
@@ -1,30 +1,40 @@
 import openai
-from config import AppConfig
 from abc import ABC, abstractmethod
 from chat.exceptions import ChatResponseGenerationError
+from config import AppConfig
 
 class Chat(ABC):
     @abstractmethod
     def generate_response(self, query: str, context: str = None) -> str:
-        """Generate a response based on the query and optional context."""
+        """Generate a response based on the query and current session's context."""
         pass
 
 class ChatOpenAI(Chat):
-    
+
     def __init__(self):
         config = AppConfig()
         openai.api_key = config.openai_api_key
-        self.history = []
+        self.history = [self._get_generate_system()]
 
-    def generate_response(self, query, context=None):
+    def _get_generate_system(self) -> dict:
+
+        return {
+            "role": "system",
+            "content": """
+                        You are an assistant that can only provide responses based on the PROVIDED CONTEXT. 
```

Comment:
Do we know for sure that this prompting works though?

Reply:
Not sure. But I believe it won't be an issue for RAG later. Because if the all the chunks have a lower similarity than the query, we just output something like 'I don't know'